source/tutorials.rst

```diff
@@ -0,0 +1,122 @@
+
+   from pytest_container import auto_container_parametrize
+
+
+   def pytest_generate_tests(metafunc):
+       auto_container_parametrize(metafunc)
+
+
+
+The above code snippet ensures that the ``auto_container`` and
+``auto_container_per_test`` fixtures work correctly (more specifically, that
+they are using the images defined in ``CONTAINER_IMAGES``).
+
+
+We can now implement the actual tests. We will create a simple smoke test where
+we install the python wheel of our project inside a container and then check
+that the installed command line utility works as expected. For that, we will
+start out with a :file:`Dockerfile`:
+
+.. code-block:: Dockerfile
+
+   FROM registry.opensuse.org/opensuse/tumbleweed
+
+   RUN zypper -n in python3-pip
+   COPY dist/*whl .
+   RUN pip install *whl
+
+
+Assuming that this :file:`Dockerfile` were in our project's root and our python
+project is built into the :file:`dist/` folder, we would build an image based on
+openSUSE Tumbleweed with our project installed into it.
+
+We can plug this :file:`Dockerfile` into ``pytest_container`` using the
+:py:class:`~pytest_container.container.DerivedContainer` class as follows:
+
+.. code-block:: python
+
+   from textwrap import dedent
+   from pytest_container import DerivedContainer
+
+
+   TW_WITH_PKG = DerivedContainer(
+       base="registry.opensuse.org/opensuse/tumbleweed",
+       containerfile=dedent("""
+       RUN zypper -n in python3-pip
+       COPY dist/*whl .
+       RUN pip install *whl
+       """)
+   )
+
+Note that we omitted the ``FROM`` line from the
+:py:attr:`~pytest_container.container.DerivedContainer.containerfile` parameter
+as ``pytest_container`` will automatically include it for us.
+
+Now we add the above snippet into a new file :file:`tests/test_cli.py` and add
+the following test function and global variable as well:
+
+.. code-block:: python
+
+   CONTAINER_IMAGES = [TW_WITH_PKG]
+
+   def test_help_works(auto_container):
+       res = auto_container.connection.run_expect([0], "my-binary --help")
+       assert "My cool project" in res.stdout
+
+
+The global variable ``CONTAINER_IMAGES`` instructs ``pytest_container`` to run
+all test functions that use the ``auto_container`` or
+``auto_container_per_test`` fixtures once for each image defined in that
+list. This allows you to have a single file with a plethora of tests that shall
+all be executed inside multiple container images and thus saves us the burden of
+manually parametrizing each of them.
+
+The test function receives a
+:py:class:`~pytest_container.container.ContainerData` instance, where we then use
+the :py:attr:`~pytest_container.container.ContainerData.connection` attribute to
+execute our projects binary. We leverage the `run_expect
+<https://testinfra.readthedocs.io/en/latest/modules.html#testinfra.host.Host.run_expect>`_
+function to execute the binary and check that its exit code is
+``0``. Afterwards, we assert that a search string is in the standard output.
```

Comment:
```suggestion
The test function receives a
:py:class:`~pytest_container.container.ContainerData` instance, where 
the :py:attr:`~pytest_container.container.ContainerData.connection` attribute is used to
execute the project's binary. The `run_expect
<https://testinfra.readthedocs.io/en/latest/modules.html#testinfra.host.Host.run_expect>`_
function is used to execute the binary and check that its exit code is
``0``. Afterwards, we assert that a search string is in the standard output.
```
_leverage_ is a stupid word.

> Afterwards, we assert that a search string is in the standard output.

Assert a search  string? What does this mean?